src/cpp/rtps/resources/ResourceEvent.cpp

```diff
@@ -179,6 +179,12 @@ void ResourceEvent::event_service()
                 current_time_ + std::chrono::seconds(1) :
                 active_timers_[0]->next_trigger_time();
 
+        auto current_time = std::chrono::steady_clock::now();
+        if (current_time > next_trigger)
+        {
+            next_trigger = current_time += std::chrono::microseconds(10);
```

Comment:
As `current_time` is not used after this line, it makes no sense to use `+=`

```suggestion
            next_trigger = current_time + std::chrono::microseconds(10);
```

Reply:
WOW! And was it compiling? Good catch.